app/src/main/java/unipd/se18/ocrcamera/PreProcessing.java

```diff
@@ -0,0 +1,369 @@
+            double[] vec = lines.get(i, 0);
+
+            //First point
+            double x1 = vec[0];
+            double y1 = vec[1];
+
+            //Second point
+            double x2 = vec[2];
+            double y2 = vec[3];
+
+            //Sum all the angle of the lines of text
+            meanAngle += Math.atan2(y2 - y1, x2 - x1);
+        }
+
+        //Calculate the meanAngle by dividing it with the number of rows
+        meanAngle /= lines.rows();
+
+        //Transform the angle in degrees
+        double degreesAngle = Math.toDegrees(meanAngle);
+        Log.i(TAG, "Mean angle=" + degreesAngle);
+        return degreesAngle;
+    }
+
+    /**
+     * Rotate the given bitmap with the given angle
+     * @param original The image that we want to corrected
+     * @param degrees The degrees of the angle we want to rotate
+     * @return the rotatedImage. If the image is null return the original image
+     * @author Thomas Porro(g1), Giovanni Fasan (g1), Oscar Garrido (g1)
+     */
+    private static Bitmap rotateBitmap(Bitmap original, double degrees) {
+
+        //TODO not correct because it returns null, maybe we should add a @NonNull
+        if(original == null){
+            return original;
+        }
+
+        //Obtain the dimen of the image
+        int width = original.getWidth();
+        int height = original.getHeight();
+
+        //Prepare the rotation matrix. The minus before degrees allows us to rotate the
+        //image in the right way
+        Matrix matrix = new Matrix();
+        matrix.preRotate((float)-degrees);
+
+        //Rotate the Bitmap and returns it
+        return Bitmap.createBitmap(original, 0, 0, width, height, matrix, true);
+
+    }
+
+
+    /**
+     * Performs the skew correction
+     * @param image The image that I want to rotate
+     * @return The image corrected
+     * @author Thomas Porro(g1), Giovanni Fasan (g1), Oscar Garrido (g1)
+     */
+    private Bitmap editSkew(Bitmap image){
+        double angle = computeSkew(image);
+        return rotateBitmap(image, angle);
+    }
+
+
+    /**
+     * Detect if the image is bright
+     * @param imageMat the image we want to detect the brightness
+     * @return 0 if image is neither too bright nor too dark,
+     *         1 if image is too bright,
+     *         2 if image is too dark.
+     * @author Thomas Porro(g1), Giovanni Fasan(g1), Leonardo Pratesi(g1)
+     */
+    private int isBright(Mat imageMat){
+        //Converts the image into a matrix
+        Mat brightnessMat = new Mat();
+
+        //Changes the format of the matrix
+        Imgproc.cvtColor(imageMat, brightnessMat, Imgproc.COLOR_RGBA2RGB);
+
+        //Obtain 3 different matrix with the 3 elemental colors
+        List<Mat> color = new ArrayList<>();
+        Core.split(brightnessMat, color);
+
+        /*Each color is multiplied with his luminance.
+          For more informarions see https://en.wikipedia.org/wiki/Relative_luminance*/
```

Comment:
If you like there is a method to include link as this in the javadoc comments, in this case it is similar to this:
@see \<a href\="[https://en.wikipedia.org/wiki/Relative_luminance](https://en.wikipedia.org/wiki/Relative_luminance)"\>Relative luminance\</a\>

Reply:
Thank you for your suggestion